libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1272,26 +1272,21 @@ void AP_InertialSensor::update(void)
     wait_for_sample();
 
     if (!_hil_mode) {
-        for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            // mark sensors unhealthy and let update() in each backend
-            // mark them healthy via _publish_gyro() and
-            // _publish_accel()
-            _gyro_healthy[i] = false;
-            _accel_healthy[i] = false;
-            _delta_velocity_valid[i] = false;
-            _delta_angle_valid[i] = false;
-        }
+        // mark sensors unhealthy and let update() in each backend
+        memset(_gyro_healthy,           0, sizeof(_gyro_healthy[0]) * INS_MAX_INSTANCES);
+        memset(_accel_healthy,          0, sizeof(_accel_healthy[0]) * INS_MAX_INSTANCES);
+        memset(_delta_velocity_valid,   0, sizeof(_delta_velocity_valid[0]) * INS_MAX_INSTANCES);
+        memset(_delta_angle_valid,      0, sizeof(_delta_angle_valid[0]) * INS_MAX_INSTANCES);
+
         for (uint8_t i=0; i<_backend_count; i++) {
             _backends[i]->update();
         }
 
         // clear accumulators
-        for (uint8_t i = 0; i < INS_MAX_INSTANCES; i++) {
-            _delta_velocity_acc[i].zero();
```

Comment:
Not calling zero here is... interesting.  What if that object held a cache of a derived value?

So long as there's a comment on the `zero()` method mentioning that places might assume a `memset(0...)` works it would be OK.

Reply:
The only members of Vector3 are <code>T x, y, z;</code>. The memset must be safe.